security/security.c

```diff
@@ -86,6 +88,37 @@ static __initconst const char * const builtin_lsm_order = CONFIG_LSM;
 static __initdata struct lsm_info **ordered_lsms;
 static __initdata struct lsm_info *exclusive;
 
+/* Create the static slots for each LSM hook */
+#define CREATE_STATIC_SLOT(NUM, NAME, RET, ...)				\
+	DEFINE_STATIC_CALL_NULL(STATIC_SLOT(NAME, NUM),			\
+				*((RET(*)(__VA_ARGS__))NULL));
+
+#define LSM_HOOK(RET, DEFAULT, NAME, ...) 				\
+	SECURITY_FOREACH_STATIC_SLOT(CREATE_STATIC_SLOT, NAME, RET, __VA_ARGS__)
+#include <linux/lsm_hook_defs.h>
+#undef LSM_HOOK
+#undef CREATE_STATIC
+
+struct security_list_static_slots security_list_static_slots
+__lsm_ro_after_init = {
+#define DEFINE_SLOT(NUM, NAME)						\
+	(struct security_static_slot) {					\
+		.key = &STATIC_CALL_KEY(STATIC_SLOT(NAME, NUM)),	\
+		.tramp = &STATIC_CALL_TRAMP(STATIC_SLOT(NAME, NUM))	\
+	},
+#define LSM_HOOK(RET, DEFAULT, NAME, ...) 				\
+	.NAME = {							\
+		.slots = {						\
+			SECURITY_FOREACH_STATIC_SLOT(DEFINE_SLOT, NAME)	\
+		},							\
+		.first_slot = INT_MAX,					\
+		.head = &security_hook_heads.NAME			\
+	},
+#include <linux/lsm_hook_defs.h>
+#undef LSM_HOOK
+#undef DEFINE_SLOT
+};
```

Comment:
We could also get rid of this struct (only keep the `first_slot` for each hook) and then use the X-macro in `lsm_init_static_slots`. This would save memory that is used only during the init, but make the code heavier.